File: 体育场馆预约管理系统/WindowsApp1/登录界面.vb
Imports System.Data.SqlClient
Public Class 登录界面
    Dim cn As SqlConnection
    Dim cm As SqlCommand

    Private Sub 登陆界面_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        '登录界面为启动页面为了让欢迎界面先出现 设置两个timer来登录界面隐藏
        欢迎界面.Show()
        Timer1.Interval = 1 '1000=1秒
        Timer1.Enabled = True
        Timer2.Interval = 3000 '1000=1秒
        Timer2.Enabled = True
    End Sub

    Private Sub Timer1_Tick(sender As Object, e As EventArgs) Handles Timer1.Tick
        Me.Hide()
        Timer1.Enabled = False
    End Sub


    Private Sub Timer2_Tick(sender As Object, e As EventArgs) Handles Timer2.Tick
        Me.Show()
        Timer2.Enabled = False
    End Sub

    Private Sub btnRegister_Click(sender As Object, e As EventArgs) Handles btnRegister.Click
        Me.Hide()
        注册界面.Show()
    End Sub

    Private Sub btnLogin2_Click(sender As Object, e As EventArgs) Handles btnLogin.Click
        If txtID.Text = "" Then
            MsgBox("请输入账号！")

        End If
        If txtPwd.Text = "" Then
            MsgBox("请输入密码！")
        End If
        Dim cnStr As String = "Data Source=DESKTOP-DSP6URK\SQLEXPRESS;Integrated Security=SSPI;Initial Catalog=changguan;"
        cn = New SqlConnection(cnStr)
        cn.Open()

        If radUser.Checked Then
            cm = New SqlCommand("select count(*) from Users where ID='" + txtID.Text + "'and Upassword='" + txtPwd.Text + "'", cn)
            Dim num1 = Convert.ToInt32(cm.ExecuteScalar())
            If (num1 > 0) Then
                cn.Close()
                MessageBox.Show("登录成功", "提示", MessageBoxButtons.OK, MessageBoxIcon.Information)
                用户界面.Show()
                Me.Visible = False
            Else
                MessageBox.Show("用户名或密码错误", "警告", MessageBoxButtons.OK, MessageBoxIcon.Information)
            End If
        ElseIf radAdmin.Checked Then
            cm = New SqlCommand("select count(*) from Users where ID='" + txtID.Text + "'and Upassword='" + txtPwd.Text + "'", cn)
            Dim num2 = Convert.ToInt32(cm.ExecuteScalar())
            If (num2 > 0) Then
                cn.Close()
                MessageBox.Show("登录成功", "提示", MessageBoxButtons.OK, MessageBoxIcon.Information)
                管理员界面.Show()
                Me.Visible = False
            Else
                MessageBox.Show("用户名或密码错误", "警告", MessageBoxButtons.OK, MessageBoxIcon.Information)
            End If

        Else
            MsgBox("请选择登录类型！")
        End If

    End Sub

End Class

File: c#体育馆预约/CS/登录界面.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using System.Linq;
using System.Drawing;
using System.Diagnostics;
using System.Threading.Tasks;
using System.Data;
using System.Xml.Linq;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports

using System.Data.SqlClient;
using WindowsApp1;
using MySql.Data.MySqlClient;

namespace WindowsApp1
{
	public partial class 登录界面
	{
		public 登录界面()
		{
			InitializeComponent();
            this.AutoScaleMode = AutoScaleMode.Font;
            //Added to support default instance behavour in C#
            if (defaultInstance == null)
				defaultInstance = this;
            
		}
		
#region Default Instance
		
		private static 登录界面 defaultInstance;
		
		/// <summary>
		/// Added by the VB.Net to C# Converter to support default instance behavour in C#
		/// </summary>
		public static 登录界面 Default
		{
			get
			{
				if (defaultInstance == null)
				{
					defaultInstance = new 登录界面();
					defaultInstance.FormClosed += new System.Windows.Forms.FormClosedEventHandler(defaultInstance_FormClosed);
				}
				
				return defaultInstance;
			}
			set
			{
				defaultInstance = value;
			}
		}
		
		static void defaultInstance_FormClosed(object sender, System.Windows.Forms.FormClosedEventArgs e)
		{
			defaultInstance = null;
		}

        #endregion
        ConDatabase ConDatabase = new ConDatabase();
        MySqlConnection con;
        MySqlCommand cm;
        
		
		public void 登陆界面_Load(object sender, EventArgs e)
		{
			//登录界面为启动页面为了让欢迎界面先出现 设置两个timer来登录界面隐藏
			欢迎界面.Default.Show();
			Timer1.Interval = 1; //1000=1秒
			Timer1.Enabled = true;
			Timer2.Interval = 3000; //1000=1秒
			Timer2.Enabled = true;
            this.WindowState = FormWindowState.Maximized;
            
        }
		
		public void Timer1_Tick(object sender, EventArgs e)
		{
			this.Hide();
			Timer1.Enabled = false;
		}
		
		
		public void Timer2_Tick(object sender, EventArgs e)
		{
			this.Show();
			Timer2.Enabled = false;
		}
		
		public void btnRegister_Click(object sender, EventArgs e)
		{
			this.Hide();
			注册界面.Default.Show();
		}
		
		public void btnLogin2_Click(object sender, EventArgs e)
		{
            //调试
            //用户界面.Default.Show();
            //this.Visible = false;
            cm = ConDatabase.OpenDatabase("select * from v_users where Uid='" + txtID.Text + "'and Upwd='" + txtPwd.Text + "'and Uidentity='管理员'" );
            con = ConDatabase.getCon();
            if (txtID.Text == "")
            {
                MessageBox.Show("请输入账号！");

            }
            else if (txtPwd.Text == "")
            {
                MessageBox.Show("请输入密码！");
            }
            else
            {
                //string conn = "server=121.36.57.112;Uid=customer;password=Summer2020;Database=summer2020";
                //con = new MySqlConnection(conn);
                //con.Open();                
                //cm = new MySqlCommand("select * from v_users where Uid='" + txtID.Text + "'and Upwd='" + txtPwd.Text + "'", con);
                var num1 = Convert.ToInt32(cm.ExecuteScalar());
                if (num1 > 0)
                {                    
                    con.Close();
                    //MessageBox.Show("登录成功", "提示", MessageBoxButtons.OK, MessageBoxIcon.Information);
                    用户界面.Default.Show();
                    this.Visible = false;
                }
                else
                {
                    MessageBox.Show("用户名或密码错误", "警告", MessageBoxButtons.OK, MessageBoxIcon.Information);
                }
            }            
        }		
	}
}
